checker/tests/lock/Issue524.java

```diff
@@ -0,0 +1,18 @@
+import org.checkerframework.checker.lock.qual.*;
+import org.checkerframework.dataflow.qual.*;
+
+//https://github.com/typetools/checker-framework/issues/524
+
+import java.util.concurrent.locks.*;
+
+// Note that this test is nondeterministic - it sometimes passes when it should fail.
+class Issue524 {
+    class MyClass { public Object field; }
+
+    void testLocalVariables() {
+        @GuardedBy("localLock") MyClass q = new MyClass();
+        @GuardedBy({}) ReentrantLock localLock = new ReentrantLock();
+        localLock.lock();
+        q.field.toString();
+    }
```

Comment:
I think this is a nice-to-have feature. I only recall seeing lone .lock() calls in definitions of higher-level lock() methods. I think the right way to do this is to have a special Lock Checker method annotation that suppresses this warning, rather than having the user add a @SuppressWarnings. I'll add a TODO in the Lock Checker - I won't have a chance to add it for this release.